include/mxnet/c_api.h

```diff
@@ -208,6 +208,24 @@ MXNET_DLL const char *MXGetLastError();
 //-------------------------------------
 // Part 0: Global State setups
 //-------------------------------------
+
+/*!
+ * \brief Check if a feature is enabled in the runtime
+ * \param feature to check mxruntime.h
+ * \param out set to true if the feature is enabled, false otherwise
+ * \return 0 when success, -1 when failure happens.
+ */
+MXNET_DLL int MXRuntimeHasFeature(const mx_uint feature, bool *out);
```

Comment:
why not have one interface that returns the complete enum and the values for each of them? doesn't seem necessary to have a capi for testing individual feature given that it can be cached in the frontend.

Reply:
Agree, I can return the enum values, but they are also implicit on the index of the strings. One of the reasons why I like having a function to test if the feature is present is that is much simpler and faster than fiddling with strings and it doesn't allocate any memory. If we remove this function then we should have two functions, features available and features enabled. I think they are functionally equivalent, there has to be some part of the code that test the feature flags anyway, exposed or not.

Reply:
As far as I understand, this function is not intended for usage within high performance loops, so I would rather prefer usability instead of optimizing for a few bytes of allocation. I'm also in favour of a dictionary with the function name as key and the status as boolean value. For HPC cases, you could still cache the results if really necessary

Reply:
This function doesn’t have to be fast.

Reply:
I don't like two functions returning strings, there's some type checking, verification happening if you have to test the enum explicitly that you don't get with two sets of strings.